holoviews/plotting/mpl/comms.py

```diff
@@ -0,0 +1,89 @@
+import uuid
+import warnings
+
+try:
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        from matplotlib.backends.backend_nbagg import CommSocket, new_figure_manager_given_figure
+except ImportError:
+    CommSocket = object
+from mpl_toolkits.mplot3d import Axes3D
+
+from ..comms import JupyterComm
+
+mpl_msg_handler = """
+target = $('#{comms_target}');
+img = $('<div />').html(data);
+target.children().each(function () {{ $(this).remove() }})
+target.append(img)
+"""
+
+mpld3_msg_handler = """
+target = $('#fig_el{comms_target}');
+target.children().each(function () {{ $(this).remove() }});
+mpld3.draw_figure("fig_el{comms_target}", data);
+"""
```

Comment:
Yes, both registration and message handling is provided by bokeh itself, which is why I'm stuck using the JupyterPushComm for now at least.